vault-core/src/main/java/org/apache/jackrabbit/vault/fs/api/FilterSet.java

```diff
@@ -239,12 +243,45 @@ public boolean covers(@NotNull String path) {
     }
 
     /**
-     * Checks if the given item is an ancestor of the root node.
+     * Checks if the given item is an ancestor of the filter's root path.
      * @param path path of the item to check
      * @return {@code true} if the given item is an ancestor
      */
     public boolean isAncestor(@NotNull String path) {
-        return path.equals(root) || root.startsWith(path + "/") || "/".equals(path);
+        boolean isAncestor = path.equals(root) || path.equals("/") || root.startsWith(path + "/");
+        log.debug("isAncestor(root={}, path={}) -> {}", root, path, isAncestor);
+        return isAncestor;
+    }
+
+    /**
+     * Matches the given path with this filter's root. If it is an ancestor, returns the name of the first
+     * path segment of the remaining filter root "below" path.
+     *
+     * @param path Path to check
+     * @return first path segment of non-matched path, or {@code null} when path not ancestor
+     */
+    public @Nullable String getDirectChildNameTowardsFilterRoot(@NotNull String path) {
+        String result = null;
+
+        String rootMatch = appendSlashIfNeeded(root);
+        String pathMatch = appendSlashIfNeeded(path);
+
+        if (rootMatch.startsWith(pathMatch)) {
+            // get filter root after matching path (will exclude leading "/"
+            String rel = rootMatch.substring(pathMatch.length());
+
+            // truncate before first "/"
+            int slashPos = rel.indexOf('/');
+            if (slashPos >= 0) {
+                rel = rel.substring(0, slashPos);
+            }
+
+            result = rel.isEmpty() ? null : rel;
+        }
+
+        log.debug("getChildNameBelowRoot(root={}, path={}) -> {}", rootMatch, pathMatch, result);
```

Comment:
This is inconsistent with the current name of the method.

Reply:
ack